src/easylink/pipeline_graph.py

```diff
@@ -312,9 +312,15 @@ def spark_is_required(self) -> bool:
     def validate_implementation_topology(
         self, nodes: list[str], metadata_steps: list[str]
     ) -> None:
-        """Check that the subgraph induced by the nodes implemented by this implementation
-        is topologically consistent with the list of metadata steps."""
+        """Validates implementation topology against intended implementation.
+
+        Check that the subgraph induced by the nodes implemented by this implementation
+        is topologically consistent with the list of steps intended to be implemented.
+        """
         subgraph = ImplementationGraph(self).subgraph(nodes)
+        # NOTE: The subgraph is not necessarily able to be topologically sorted
```

Comment:
@patricktnast Does this note jive with your understanding? We should talk about this. I'm confused why an nx graph isn't just _always_ sorted. And, related, I'm confused why I am unable to access the `is_sorted` attr sometimes (it's not False, it's actually not set)